src/Dash.NET.Suave/App.fs

```diff
@@ -0,0 +1,252 @@
+  static member mapIndex (mapping: IndexView -> IndexView) (app:DashApp) =
+    { app with 
+        Index = mapping app.Index 
+    }
+
+  /// Returns a new DashApp with the given css source links appended to the original IndexView's css register as link tags
+  static member appendCSSLinks (hrefs:seq<string>) (app:DashApp) =
+    let tags = 
+      hrefs
+      |> Seq.map (fun href ->
+          link ["rel", "stylesheet"; "href", href ; "crossorigin", " "]
+      )
+    app
+    |> DashApp.mapIndex (IndexView.appendCSSLinks tags)
+
+  /// Returns a new DashApp with the given script source links appended to the original IndexView's script register as script tags
+  static member appendScripts (sources:seq<string>) (app:DashApp) =
+    let tags = 
+      sources
+      |> Seq.map (fun source ->
+          script [ "type", "application/javascript"; "crossorigin", " "; "src", source] []
+      )
+    app
+    |> DashApp.mapIndex (IndexView.appendScripts tags)
+
+  /// Returns a new DashApp with the original Layout replaced by the given Layout 
+  static member withLayout (layout:DashComponent) (app:DashApp) =
+      { app with 
+          Layout = layout
+      }
+
+  /// Returns a new DashApp with the given callback added to the original Callback register
+  static member addCallback (callback: Callback<'Function>) (app: DashApp) =
+
+    //To-Do: Maybe use copy utility for all direct calls to underlying DynamicObjs.
+    { app with
+        Callbacks =
+          app.Callbacks
+          |> CallbackMap.registerCallback callback
+    }
+
+  /// Returns a new DashApp with the given callbacks added to the original Callback register
+  static member addCallbacks (callbacks: seq<(Callback<'Function>)>) (app: DashApp) =
+    //To-Do: Maybe use copy utility for all direct calls to underlying DynamicObjs.
+    { app with
+        Callbacks =
+            callbacks
+            |> Seq.fold (fun cMap cHandler ->
+                cMap
+                |> CallbackMap.registerCallback cHandler) app.Callbacks
+    }
+
+  /// Returns the given DashApp's index DOM
+  static member getIndexHTML (app:DashApp) =
+    app.Index |> IndexView.toHTMLComponent
+
+  /// Returns the DasApp as a http handler to use as Giraffe middleware in an asp.netcore pipeline. 
+  ///
+  /// This application will provide the following endpoints:
+  ///
+  /// GET / -> serves the DashApps IndexView (This is also where the Das renderer will render the layout)
+  ///
+  /// GET /_dash-layout -> serves the DashApp's layout as JSON
+  ///
+  /// GET /_dash-dependencies -> serves the serialized callback handlers registered in the DashApp's callback register as JSON
+  ///
+  /// GET /_reload-hash -> Not implemented, returns empty JSON object
+  ///
+  /// POST /_dash-update-component -> handles callback requests and returns serialized callback JSON responses.
+
+  static member toWebPart (app:DashApp) : WebPart =
```

Comment:
It looks to me like the only thing we need separately in Dash.NET.Giraffe and Dash.NET.Suave eventually is the `toWebPart` / `toHttpHandler` (and run, see above) functions right? if so, we can keep all of the functions above in Dash.NET core and just add these as extension methods